src/main/c/Objects/pyjmethod.c

```diff
@@ -179,7 +179,6 @@ int PyJMethod_CheckArguments(PyJMethodObject* method, JNIEnv *env,
                                 method->parameters, parampos);
 
         if (process_java_exception(env) || !paramType) {
-            match = 0;
```

Comment:
I think we should set matchTotal to -1 and document in the header that negative return values indicate errors.